code/modules/research/xenobiology/xenobio_camera.dm

```diff
@@ -109,72 +122,93 @@
-				monkeys++
-				qdel(storage_item)
-		if(loaded)
-			to_chat(user, span_notice("You fill [src] with the monkey cubes stored in [used_item]. [src] now has [monkeys] monkey cubes stored."))
-			xeno_hud.on_update_hud(LAZYLEN(stored_slimes), monkeys, max_slimes)
-		return
-
-	if(istype(used_item, /obj/item/slimepotion/slime))
-		var/replaced = FALSE
-		if(user && !user.transferItemToLoc(used_item, src))
-			return
-		if(!QDELETED(current_potion))
-			current_potion.forceMove(drop_location())
-			replaced = TRUE
-		current_potion = used_item
-		xeno_hud.update_potion(current_potion)
-		to_chat(user, span_notice("You load [used_item] in the console's potion slot[replaced ? ", replacing the previous" : ""]."))
-
-		return
-
-	..()
-
-/obj/machinery/computer/camera_advanced/xenobio/multitool_act(mob/living/user, obj/item/multitool/used_multitool)
+/obj/machinery/computer/camera_advanced/xenobio/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
 	. = ..()
-	if (istype(used_multitool) && istype(used_multitool.buffer,/obj/machinery/monkey_recycler))
-		to_chat(user, span_notice("You link [src] with [used_multitool.buffer] in [used_multitool] buffer."))
-		connected_recycler = used_multitool.buffer
-		connected_recycler.connected += src
-		return TRUE
-
-/*
-Boilerplate check for a valid area to perform a camera action in.
-Checks if the AI eye is on a valid turf and then checks if the target turf is xenobiology compatible
-Due to keyboard shortcuts, the second one is not necessarily the remote eye's location.
-*/
-/obj/machinery/computer/camera_advanced/xenobio/proc/validate_area(mob/living/user, mob/eye/camera/remote/xenobio/remote_eye, turf/open/target_turf)
-	if(!GLOB.cameranet.checkTurfVis(remote_eye.loc))
-		to_chat(user, span_warning("Target is not near a camera. Cannot proceed."))
+	if(.)
+		return .
+
+	if(istype(tool, /obj/item/slimepotion/slime))
+		return slimepotion_act(user, tool)
+	if(istype(tool, /obj/item/food/monkeycube))
+		return monkeycube_act(user, tool)
+	if(istype(tool, /obj/item/storage/bag) || istype(tool, /obj/item/storage/box/monkeycubes))
+		return storage_act(user, tool)
+	return NONE
+
+/// Handles inserting a slime potion into the console, potentially swapping out an existing one.
+/obj/machinery/computer/camera_advanced/xenobio/proc/slimepotion_act(mob/living/user, obj/item/slimepotion/slime/used_potion)
+	if(!user.transferItemToLoc(used_potion, src))
+		balloon_alert(user, "can't insert!")
+		return ITEM_INTERACT_BLOCKING
+
+	if(!QDELETED(current_potion))
+		try_put_in_hand(current_potion, user)
+		balloon_alert(user, "swapped")
+	else
+		balloon_alert(user, "inserted")
+
+	current_potion = used_potion
+	xeno_hud.update_potion(current_potion)
+	return ITEM_INTERACT_SUCCESS
+
+/// Handles inserting a monkey cube into the console.
+/obj/machinery/computer/camera_advanced/xenobio/proc/monkeycube_act(mob/living/user, obj/item/food/monkeycube/used_cube)
+	stored_monkeys += 1
+	balloon_alert(user, "inserted")
```

Comment:
Does this fire once for every monkey cube you put in? What if you put in a full science-bag of cubes?